BAZProject/Utilities/MovieAPI.swift

```diff
@@ -7,78 +7,112 @@
+    var getEndpointUrl: String {
         switch self {
         case .upcoming:
-            return "\(urlBase)/movie/upcoming?api_key=\(apiKey)&language=es&region=MX&page=1"
-        case .trending(let page):
-            return "\(urlBase)/trending/movie/day?api_key=\(apiKey)&language=es&region=MX&page=\(page)"
-        case .nowPlaying(let page):
-            return "\(urlBase)/movie/now_playing?api_key=\(apiKey)&language=es&region=MX&page=\(page)"
-        case .popular(let page):
-            return "\(urlBase)/movie/popular?api_key=\(apiKey)&language=es&region=MX&page=\(page)"
-        case .topRated(let page):
-            return "\(urlBase)/movie/top_rated?api_key=\(apiKey)&language=es&page=\(page)&region=MX"
-        case .keyword(let query):
-            return "\(urlBase)/search/keyword?api_key=\(apiKey)&language=es&query=\(query)"
-        case .searchMovie(let query, let page):
-            return "\(urlBase)/search/movie?api_key=\(apiKey)&language=es&page=\(page)&query=\(query)"
-        case .reviews(let movieId):
-            return "\(urlBase)/movie/\(movieId)/reviews?api_key=\(apiKey)&language=es"
-        case .similar(let movieId):
-            return "\(urlBase)/movie/\(movieId)/similar?api_key=\(apiKey)&language=es"
-        case .recommendations(let movieId):
-            return "\(urlBase)/movie/\(movieId)/recommendations?api_key=\(apiKey)&language=es"
+            return "/movie/upcoming"
+        case .trending:
+            return "/trending/movie/day"
+        case .nowPlaying:
+            return "/movie/now_playing"
+        case .popular:
+            return "/movie/popular"
+        case .topRated:
+            return "/movie/top_rated"
+        case .keyword:
+            return "/search/keyword"
+        case .searchMovie:
+            return "/search/movie"
+        case .reviews:
+            return "/reviews"
+        case .similar:
+            return "/similar"
+        case .recommendations:
+            return "/recommendations"
+        case .creditMovie:
+            return "/credits"
         }
     }
 }
 
 final class MovieAPI {
-    private let imgBaseUrl: String = "https://image.tmdb.org/t/p/w500"
-    public var getDataMovies: Codable?
-    public var dictionaryUrls: [URLApi]?
-    static let movieAPISharedInstance = MovieAPI()
+    static private let imgBaseUrl: String = "https://image.tmdb.org/t/p/w500"
+    static private var apiKey: String = "f6cd5c1a9e6c6b965fdcab0fa6ddd38a"
+    static private var urlBase: String = "https://api.themoviedb.org/3"
+    
     /**    func to help to get Data of apis
      - Parameter url: url of api
      
      */
-    func getApiData(from url:URLApi, handler: @escaping (Data) -> Void){
-       guard let url = URL(string: url.URL) else{return}
+    static func getApiData(from url:URLApi, handler: @escaping (Data) -> Void) {
+        guard let url = URL(string: "\(urlBase)\(url.getEndpointUrl)?api_key=\(apiKey)&language=es&region=MX&page=1") else { return }
```

Comment:
You should consider a helper for the URLs construction or even a method in this class. You could use `URLComponents` to create the URL and avoid having an string like this one